Pipeline/PackageBuildOutputs/PackageBuildOutputs.groovy

```diff
@@ -0,0 +1,318 @@
+			if (stderr){
+				println ("*! stderr : $stderr")
+				println ("*! stdout : $stdout")
+			}
+
+		}
+	}
+}
+
+def tarFile = new File("$props.workDir/${tarFileName}")
+
+println("** Creating tar file at $tarFile.")
+// Note: https://www.ibm.com/docs/en/zos/2.4.0?topic=scd-tar-manipulate-tar-archive-files-copy-back-up-file
+// To save all attributes to be restored on z/OS and non-z/OS systems : tar -UX
+def processCmd = [
+	"sh",
+	"-c",
+	"tar cUXf $tarFile *"
+]
+
+def rc = runProcess(processCmd, tempLoadDir)
+assert rc == 0 : "Failed to package"
+
+//Package BuildReport.json to carry BuildInfo including deployTypes etc.
+println("** Adding BuildReport.json to $tarFile.")
+processCmd = [
+	"sh",
+	"-c",
+	"tar rUXf $tarFile BuildReport.json"
+]
+
+rc = runProcess(processCmd, new File(props.workDir))
+assert rc == 0 : "Failed to append BuildReport.json"
+
+println ("** Package successfully created at $tarFile.")
+
+//Set up the artifactory information to publish the tar file
+if (props.publish && props.publish.toBoolean()){
+	// Configuring ArtifactoryHelper parms
+	def String remotePath = (props.versionName) ? (props.versionName + "/" + tarFileName) : (tarFileLabel + "/" + tarFileName)
+	def url = new URI(props.get('artifactory.url') + "/" + props.get('artifactory.repo') + "/" + remotePath ).normalize().toString() // Normalized URL
+
+	def apiKey = props.'artifactory.user'
+	def user = props.'artifactory.user'
+	def password = props.'artifactory.password'
+	def repo = props.get('artifactory.repo') as String
+
+	//Call the ArtifactoryHelpers to publish the tar file
+	def scriptDir = new File(getClass().protectionDomain.codeSource.location.path).parent
+	File artifactoryHelpersFile = new File("$scriptDir/ArtifactoryHelpers.groovy")
+	Class artifactoryHelpersClass = new GroovyClassLoader(getClass().getClassLoader()).parseClass(artifactoryHelpersFile)
+	GroovyObject artifactoryHelpers = (GroovyObject) artifactoryHelpersClass.newInstance()
+
+	println ("** Uploading package to Artifactory $url.")
+	artifactoryHelpers.upload(url, tarFile as String, user, password, props.verbose.toBoolean() )
+}
+
+/**
+ * parse data set name and member name
+ * @param fullname e.g. BLD.LOAD(PGM1)
+ * @return e.g. (BLD.LOAD, PGM1)
+ */
+def getDatasetName(String fullname){
+	def ds,member;
+	def elements =  fullname.split("[\\(\\)]");
+	ds = elements[0];
+	member = elements.size()>1? elements[1] : "";
+	return [ds, member];
+}
+
+/**
+ * run process
+ */
+def runProcess(ArrayList cmd, File dir){
+	if (props.verbose && props.verbose.toBoolean()) println "executing $cmd: "
+	StringBuffer response = new StringBuffer()
+	StringBuffer error = new StringBuffer()
+
+	// execute cmd
+	def p = cmd.execute(null, dir)
+
+	p.waitForProcessOutput(response, error)
+	println(response.toString())
+
+	def rc = p.exitValue();
+	if(rc==0){
+
+	}else{
+		println("*! Error executing $cmd \n" + error.toString())
+		//System.exit(1)
+	}
+	return rc
```

Comment:
Can you simplify?
if (rc!=0) {
println("*! Error executing $cmd \n" + error.toString())
}